### ISSUE TITLE
Reading clidata files, through Excel

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#2294]
Clidata is a second data management system, from the Czech republic and used by a range of Met services in Africa including Ghana.

I asked Patrick for some data to use to test R-Instat.
a) he first supplied me with the station data and a version of the daily data, but with values "going across", i.e. 1 to 31 for days of the month.
b) I asked whether he could alternatively take the data out of clidata and it came in the attached form.  It is in an old version of Excel - I think 5/95.  I have not checked whether it could come in an ASCII form.
c) I tried reading it in as it stands.  I used the option to omit the first 5 rows.  It looked OK but gave an error.  Perhaps this is because 3 of the columns had no name?  But I could not understand the error.
d) I then took the attached file and copied the names down to the right row - in Excel.  I saved in the latest form of Excel.
e) I then tried to read in again.  It didn't allow me to preview.  I experimented and it seems it cannot cope (in the latest Excel) with leaving the 5 lines.
f) So I eliminated the 5 lines in the Excel file and then tried again.  It read fine.  Patrick was impressed.  (But I noticed that the year, month, day columns are read as text, so there could be a problem somewhere.  And the date column is not recognised as such.
g) Then I added the station file.  Here is a definite bug in our system.  In both cases, the sheet is called Sheet1.  But it comes from a different file.  However, in R-Instat it overwrites if the sheet name is the same.
h) I renamed the station sheet, and now imported.
i) Now I want to check the Year/Month/Day columns.  It would be good in the filter to be able to select all non-numeric rows?  We need this sort of thing also in our find-replace.  In Excel I would do a filter and look in the list at the top.  I now find there is nothing non-numeric.  I wonder why it has imported as text?  I was able to check that in Excel, and then also in R-Instat by making them into a factor.
j) Now I used the Make Date dialogue.  It worked fine.  Just a small suggestion here, to add their format into our list, with full-stop as the separator, namely "%d.%m.%Y".

**Comment by rdstern** [comment]
I think the name in the column of this file may be causing some serious issues.  The name of the column is “Eg.gh.id” – don’t ask me why!.  I was processing the Ghana data fine, until I made this column into a factor.  Then I went to the Factor > Level/Labels dialogue and it didn't give me any levels for that column.  Now it also doesn't give me the levels for the year column in the same frame either.  

When I do Factor > Recode and move to that column it gives an error as follows:
---------------------------
Error running R command(s)
---------------------------
Error in self$get_data_objects(data_name) :  not found
The error occurred in attempting to run the following R command(s):
convert_to_character_matrix(data=InstatDataObject$get_factor_data_frame(data_name="", col_name="Eg.gh.id"))
When I go to a different sheet the commands work OK.
But they don’t work for any factor in that sheet.

I have now changed the name to “Station” and the problem remains.  I find it interesting that the Levels/Labels dialogue is working fine with other sheets, but is going wrong – and now giving that error above with any7 factor column in that sheet.
I deleted the sheet – that was really slow.  If it really takes this much time there should be an indication that R-Instat is working.
I now reopened the sheet and renamed the offending column immediately to “Station”.  Then I went back to the levels/Labels dialogue.  It still gave me an error – now I think for a different reason.  When I got into the dialogue it still had “Month” from before.  But that sheet had been deleted and I had re-opened, and not (yet) made Month again into a factor column.

**Comment by dannyparsons** [comment]
@rdstern good to test with this data again?

**Comment by dannyparsons** [comment]
@rdstern could you attach or send the data so we can test this again? Is this still a 0.4.1 or could it be 0.4.2 now?

**Comment by rdstern** [comment]
[Gh_multiple data_15112016.zip](https://github.com/africanmathsinitiative/R-Instat/files/1204734/Gh_multiple.data_15112016.zip)
These are the data files.  They now import fine - both of them.
One needs to omit the first 5 lines, which is now an option.  Some columns also don't have names, so that's also now OK.

Recently I have found the importing to be very robust - and this is another example.

The one problem - at least on my machine - reported elsewhere.  The first file (wide one took a long time to preview, i.e. the message to wait appeared at least 5 times.  I can accept this.  BUT then, when I went to the second file I had (of course) to go through all this again even though I didn't want that file anymore!
